Formula/navi.rb

```diff
@@ -11,14 +11,18 @@ class Navi < Formula
     sha256 "64b6dac7d4c689ab6aa5c88ef58369249af2d9a6ac44f56769685569a673dd56" => :high_sierra
   end
 
+  depends_on "rust" => :build
   depends_on "fzf"
 
   def install
-    libexec.install Dir["*"]
-    bin.write_exec_script(libexec/"navi")
+    libexec.install Dir["cheats/*"]
+    libexec.install Dir["shell/*"]
+    system "cargo", "install", "--root", prefix, "--path", "."
+    mv "#{bin}/navi", "#{libexec}/navi"
+    ln_s "#{libexec}/navi", "#{bin}/navi"
   end
 
   test do
-    assert_equal version, shell_output("#{bin}/navi --version")
+    assert_match "navi " + version, shell_output("#{bin}/navi --version")
```

Comment:
If we can improve the test a little bit, that would be great.

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Done!